upgrade-notes/backend_20241205_151830.md

```diff
@@ -0,0 +1,7 @@
+#### Admin Grid performance improvements ([#2460](https://github.com/shopsys/shopsys/pull/{pullRequestId}))
+
+- Modified grid so that adding NULL FIRST / NULLS LAST is optional now, because it is much more time-consuming.
+- QueryBuilderWithRowManipulatorDataSource called callback for each item separately without the possibility of working with all items. Modified callback so that instead of calling ID = X for each item, ID IN (X, X, X) could be called only once. See example implementation in ProductControllet. In Callback function it is possible to request the entire result from query as second parameter.
```

Comment:
ProductControllet -> ProductController

However, if you refactor the code, the example usage will be probably in the new ProductGridFactory :slightly_smiling_face: